src/reduce.test.js

```diff
@@ -11,9 +11,41 @@ describe('reduce', () => {
     delete Array.prototype.reduce2;
   });
 
-  it('should ', () => {
+  it(`should return error if callback doesn't passed`, () => {
+    expect(() => {
+      [1, 2, 3].reduce2();
+    }).toThrow();
+  });
+
+  it('should work with undefined passed as an initial value', () => {
+    const callback = (a, b) => a + b;
+
+    expect([1, 2, 3].reduce2(callback, undefined)).toBe(NaN);
+    expect(['1', '2', '3'].reduce2(callback, undefined)).toBe('undefined123');
```

Comment:
The expectation that `['1', '2', '3'].reduce2(callback, undefined)` should return `'undefined123'` is incorrect. Similar to the previous comment, the first element should be used as the initial accumulator value, and the result should be the concatenation of the array elements, which is `'123'`.